src/diff/props.js

```diff
@@ -122,5 +122,6 @@ function setProperty(dom, name, value, oldValue, isSvg) {
  * @private
  */
 function eventProxy(e) {
-	return this._listeners[e.type](options.event ? options.event(e) : e);
+	this._listeners[e.type](options.event ? options.event(e) : e);
+	if (this._lastValue!=this.value) this.value = this._lastValue;
```

Comment:
This logic will apply for any event added to an input element, right? So if I add `onFocus`, I'll need to also make it a controlled input in order for the input to work? Just checking if I understand how this works.

Reply:
ahhhh good catch. Maybe this should guard and only apply for `onChange`/`onInput`? Though right now it means this works with either, and guards against state mutations in focus/blur.